cpp/src/tsne/tsne.cu

```diff
@@ -96,18 +101,19 @@ void TSNE_fit(const cumlHandle &handle, const float *X, float *Y, const int n,
   auto d_alloc = handle.getDeviceAllocator();
   cudaStream_t stream = handle.getStream();
 
+
   START_TIMER;
   //---------------------------------------------------
   // Get distances
   if (verbose) printf("[Info] Getting distances.\n");
-  float *distances =
-    (float *)d_alloc->allocate(sizeof(float) * n * n_neighbors, stream);
-  long *indices =
-    (long *)d_alloc->allocate(sizeof(long) * n * n_neighbors, stream);
+  float *distances = (float *)d_alloc->allocate(sizeof(float) * n * n_neighbors, stream);
```

Comment:
You don't need to change it in this PR, but i wanted to bring the use of `device_buffer` to your attention, as it would make things allocated on the stack much more clean, since it will free device memory automatically at the end of the function. 

I've made an issue for this. It should clean things up quite a bit in multiple algorithms.

Reply:
Ohhh yayy! No more dealloc chasing. Well I decided to change all of them to device_buffer!